src/django_idom/templatetags/idom.py

```diff
@@ -34,17 +35,29 @@ def component(dotted_path: str, **kwargs: Any):
         </body>
         </html>
     """
-    _register_component(dotted_path)
-
+    component = _register_component(dotted_path)
+    uuid = uuid4().hex
     class_ = kwargs.pop("class", "")
-    json_kwargs = json.dumps(kwargs, separators=(",", ":"))
+    kwargs.pop("key", "")  # `key` is effectively useless for the root node
+
+    # Store the component's args/kwargs in the database if needed
+    # This will be fetched by the websocket consumer later
+    try:
+        if func_has_params(component, *args, **kwargs):
+            params = ComponentParamData(args, kwargs)
+            model = models.ComponentParams(uuid=uuid, data=pickle.dumps(params))
+            model.full_clean()
+            model.save()
+    except TypeError as e:
+        raise TypeError(
+            f"The provided parameters are incompatible with component '{dotted_path}'."
+        ) from e
 
     return {
         "class": class_,
         "idom_websocket_url": IDOM_WEBSOCKET_URL,
         "idom_web_modules_url": IDOM_WEB_MODULES_URL,
-        "idom_ws_max_reconnect_timeout": IDOM_WS_MAX_RECONNECT_TIMEOUT,
-        "idom_mount_uuid": uuid4().hex,
-        "idom_component_id": dotted_path,
-        "idom_component_params": urlencode({"kwargs": json_kwargs}),
+        "idom_reconnect_max": IDOM_RECONNECT_MAX,
+        "idom_mount_uuid": uuid,
```

Comment:
Just as a final check, could an attacker leverage this UUID in any way? For example, if credentials are passed via this mechanism, could an attacker attempt to re-use this UUID somehow?

Reply:
Yes, but the same vulnerability exists for session cookies and CSRF tokens as well. The defense mechanism comes from the fact that the uuid has enough entropy to avoid random guessing. The only ways of abuse would be brute force or man-in-the-middle attacks, both of which can be mitigated by proper webserver/firewall config.

Reply:
In the future we could potentially make this more secure through public/private key pairs (similar to SSL) but that gets into the territory of encryption, which should be handled in a separate PR.